# phylo_ABC_ATPases


# Characterization of ABC ATPases associated with immune function in prokaryotes.

**Authors**  
- Natalya Kichigina
- Alena Koshkareva

**Supervisors**  
- Oksana Kotovskaya (Skolkovo Institute of Science and Technology)

 ### Repository structure
```
ğŸ“ phylo-ABC-ATPases/
â”œâ”€â”€ ğŸ“„ README.md 
â”œâ”€â”€ ğŸ“„ environment.yml 
â”œâ”€â”€ ğŸ“ data/
â”‚   â”œâ”€â”€ ğŸ“ raw/
â”‚   â”‚   â”œâ”€â”€ ğŸ“ articles/
â”‚   â”‚   â”œâ”€â”€ ğŸ“ from_lab/
â”‚   â”‚   â”œâ”€â”€ ğŸ“ psi_blast/ 
â”‚   â”‚   â”‚   â””â”€â”€ ğŸ“ special_case/
â”‚   â””â”€â”€ ğŸ“ processed/
â”‚       â”œâ”€â”€ ğŸ“ 01_fasta_seq/
â”‚       â”œâ”€â”€ ğŸ“ 02_clustered_seq/  
â”‚       â”œâ”€â”€ ğŸ“ 03_aligned_seq/  
â”‚       â”œâ”€â”€ ğŸ“ 04_trimmed_seq/
â”‚       â””â”€â”€ ğŸ“ 05_seq_for_tree/ 
â”œâ”€â”€ ğŸ“ results/
â”‚   â”œâ”€â”€ ğŸ“ trees/  
â”‚   â”‚   â””â”€â”€ ğŸ“„ fasttree.nwk  
â”‚   â”‚    
â”‚   â”‚    
â”‚   â”œâ”€â”€ ğŸ“ alphafold_models/ 
â”‚   â””â”€â”€ ğŸ“„ result_sources_ID.csv
â””â”€â”€ ğŸ“ scripts/
    â”œâ”€â”€ ğŸ“„ 01.1_lab_data_cleanup.sh   
    â”œâ”€â”€ ğŸ“„ 01.2_csv_to_fasta_PARIS.ipynb
    â”œâ”€â”€ ğŸ“„ 02.2_download_ncbi_seqs.ipynb 
    â”œâ”€â”€ ğŸ“„ 03.1_cluster_seqs.sh 
    â”œâ”€â”€ ğŸ“„ 03.2_cluster_seqs.sh 
    â”œâ”€â”€ ğŸ“„ 04_align_seqs.sh  
    â”œâ”€â”€ ğŸ“„ 05.1_filter_walker_a&b.ipynb
    â”œâ”€â”€ ğŸ“„ 05.2_statistics_number_seq_by_source.ipynb
    â”œâ”€â”€ ğŸ“„ 07.2_draw_tree.R
    â””â”€â”€ ğŸ“„ 09_alphafold_web.md  
```

### Introduction

 ATPases are widespread enzymes that use the energy of ATP hydrolysis to change conformation. These conformational changes are usually transferred to additional domains in proteins to perform some function in the cell. For example, ABC ATPases perform a transmembrane transport function and SMC ATPases are involved in maintaining chromosome organisation. ABC ATP-ases are a group of ATPases that have two conserved domains Walker A and Walker B and lack a domain called the arginine finger. 
A portion of ABC ATP-ases have immune functions in prokaryotes. In recent years, a large number of ABC ATPases have been discovered in the field of bacterial immunity that are associated with the response to phage infection (AriA in Paris and others). 
And in this project, we would like to establish phylogenetic relationships between different groups of immune ABC ATPases to better understand the evolutionary processes in this group of ATPases.


### Data
Data. The data were obtained from three sources: firstly, literature data with already known sequences of immune ABC ATPases, secondly, protein sequences from immune proteins of bacteria known to harbour ABC-ATPase domains (e.g. PrrC, RloC). GenBank Database. predicted with PADLOC v.2.0.0. available in the National Centre for Biotechnology Information (NCBI) database, and the last source homologues of proteins studied experimentally using the Position Specific Iterated BLAST web. The final size of the assembled dataset was about 50 thousand sequences.

### workflow
1 step.
The entire protein dataset was then clustered at 70% similarity and 70% minimum coverage using MMseqs2 (17.b804f)

2 step. 
The resulting sequences were aligned using Clustal Omega (v.1.2.4)

3 step.
Alignment trimming using the ClipKIT(2.3.0)

4 step.
Filtering for the presence of two domain areas: Walker A, Walker B


5 step.

Tree construction by the FastTree(2.1.11) program (maximum-likelihood NNIs).